src/Symfony/Component/Messenger/Transport/Schedule/ScheduleTransport.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Schedule;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Exception\LogicException;
+use Symfony\Component\Messenger\Exception\TransportException;
+use Symfony\Component\Messenger\Stamp\ScheduledStamp;
+use Symfony\Component\Messenger\Transport\TransportInterface;
+
+class ScheduleTransport implements TransportInterface
+{
+    private ScheduleInterface $schedule;
+    private array $stamps;
+
+    public function __construct(ScheduleInterface $scheduler)
```

Comment:
```suggestion
    public function __construct(
        private ScheduleInterface $schedule,
    )
```

I've also renamed scheduler to schedule.